lib/core/network/token_interceptor.dart

```diff
@@ -7,18 +7,23 @@ import 'package:givt_app/features/auth/models/session.dart';
 import 'package:givt_app/features/auth/repositories/auth_repository.dart';
 import 'package:http_interceptor/http_interceptor.dart';
 import 'package:shared_preferences/shared_preferences.dart';
+import 'package:uuid/uuid.dart';
 
 class TokenInterceptor implements InterceptorContract {
   @override
   Future<BaseRequest> interceptRequest({required BaseRequest request}) async {
+    final correlationId = const Uuid().v4();
```

Comment:
Just to be sure, we want a unique correlation ID per request? (i think we do but just checking)

Reply:
yes :) 
In this way we can track the call from frontend through the entire backend system